### DISCUSSION TITLE
[feature request] Create API to determine a repo's default workflow permissions.

### DISCUSSION THREAD

**Discussion by pnacht** [community/community#38799] [Apps, API and Webhooks] [answered]
### Select Topic Area

Product Feedback

### Body

Hi, I'm Pedro and I work with Google and the OpenSSF to improve the supply-chain security of open-source projects. 

When evaluating whether a project's workflows are secure, an important factor is whether they run with minimal permissions.

This can be straightforward if workflows declare read-only top-level permissions (or `permissions: {}`). However, this very often isn't the case; projects often either only declare job-level permissions or sometimes simply don't declare any, using the defaults.

By default, workflows run with permissive (write-all) permissions, but maintainers can change this to read-only in Settings > Actions > General > Workflow Permissions. This makes determining whether a `permissions:`-less workflow is actually dangerous much more complex.

Given that other security-adjacent repo settings are made available via API (i.e. branch protection rules), exposing this information as well would also be a very useful data-point.

**Comment by ossanna16** [comment]
Hi there @pnacht and welcome to our community! Thank you for the feedback!

**Comment by pnacht** [comment, marked as answer]
This request has been handled! See https://docs.github.com/en/rest/actions/permissions#get-default-workflow-permissions-for-a-repository.
